app/services/qa/pagination_service.rb

```diff
@@ -0,0 +1,563 @@
+      end
+
+      # @return [Boolean] true if the request specifies the page offset; otherwise, false
+      def page_offset_specified?
+        request_params.keys.include?("page_offset") || request_params.keys.include?("startRecord")
+      end
+
+      # @return [Integer] The page offset as specified in the request_params, nil if invalid.
+      # @note The page offset can be specified with page_offset (preferred) or
+      #       startRecord (deprecated, supported for backward compatibility with
+      #       linked_data module pagination).
+      def requested_page_offset
+        return @requested_page_offset if @requested_page_offset.present?
+        @requested_page_offset = (request_params['page_offset'] || request_params['startRecord'])
+      end
+
+      # @return [Integer] The first record to include in the response data as
+      #     requested as long as it is an Integer; otherwise, returns nil.
+      def validated_request_page_offset
+        @page_offset_error = false
+        offset = Integer(requested_page_offset)
+        return offset if offset == 1
+        @page_offset_error = ERROR_OUT_OF_RANGE_TOO_SMALL if offset < 1
+        @page_offset_error = ERROR_OUT_OF_RANGE_TOO_LARGE if offset > total_num_found
+        offset
+      rescue ArgumentError
+        @page_offset_error = ERROR_NOT_INTEGER
+        nil
+      end
+
+      # @return [Integer] The requested maximum number of results to return (default=DEFAULT_PAGE_LIMIT | ALL)
+      # @see #default_page_limit
+      # @see DEFAULT_PAGE_LIMIT
+      def page_limit
+        return @page_limit if @page_limit.present?
+        return @page_limit = default_page_limit unless page_limit_specified?
+        @page_limit = validated_request_page_limit || default_page_limit
+      end
+
+      # @return [Boolean] true if the request specifies the page limit; otherwise, false
+      def page_limit_specified?
+        request_params.keys.include?("page_limit") || request_params.keys.include?("maxRecords")
+      end
+
+      # @return [Integer] The max number of records for a page as specified in the request_params.
+      # @note The page size limit can be specified with page_limit (preferred) or
+      #       maxRecords (deprecated, supported for backward compatibility with
+      #       linked_data module pagination).
+      def requested_page_limit
+        return @requested_page_limit if @requested_page_limit.present?
+        @requested_page_limit ||= (request_params['page_limit'] || request_params['maxRecords'])
+      end
+
+      # @return [Integer] The max number of records to include in response data as
+      #     requested as long as it is a positive Integer; otherwise, returns nil.
+      def validated_request_page_limit
+        @page_limit_error = false
+        limit = Integer(requested_page_limit)
+        @page_limit_error = ERROR_OUT_OF_RANGE_TOO_SMALL if limit < 1
+        limit.positive? ? limit : nil
+      rescue
```

Comment:
Match the change above to only rescue specific errors.
```suggestion
      rescue ArgumentError
```